gatling-app/src/main/scala/io/gatling/app/Gatling.scala

```diff
@@ -48,11 +52,11 @@ object Gatling {
 
     implicit val configuration = GatlingConfiguration.load(overrides)
 
-    new Gatling(selectedSimulationClass).start.code
+    new Gatling((v) => v.newInstance())(selectedSimulationClass).start.code
   }
 }
 
-private[app] class Gatling(selectedSimulationClass: SelectedSimulationClass)(implicit configuration: GatlingConfiguration) {
+private[app] class Gatling(simulationFactory: SimulationFactory = _.newInstance())(selectedSimulationClass: SelectedSimulationClass)(implicit configuration: GatlingConfiguration) {
```

Comment:
This class is package private. Users are only supposed to use factories in the companion. Provide a public API.